enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+
+## Open Questions [optional]
+
+> WIP/TODO
+
+- Q: Is the annotation name be changed to include "nlb" in the name? There is no similar annotation in ALBC.
+
+- Q: Is it required to create a KEP to the CCM changes?
+
+- Q: Does the CIO recreates the service when the managed flag is added? Does CCM requires to recreate the NLB when annotation is added?
+
+## Test Plan
+
+> WIP/TBReviewed
+
+**cloud-provider-aws (CCM):**:
+
+- e2e service Load Balancer type NLB with Security Groups (SG) needs to be implemented in the CCM component (upstream)
+
+- Implement e2e tests for service Load Balancer type NLB with the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` annotation. These tests should verify:
+  - The creation of an NLB with a newly created and associated Security Group.
+  - The correct configuration of Ingress rules in the Security Group based on the service ports and loadBalancerSourceRanges.
+  - The configuration of Egress rules, ensuring they are appropriately restricted.
+  - The deletion of the Security Group when the service is deleted.
+  - The default NLB provisioning behavior when the annotation is absent.
+  - The behavior when the annotation is set to true on a non-NLB service type (should result in a warning or error).
+
+**Cluster Ingress Operator (CIO)**:
+
+- Implement e2e tests to verify that the CIO correctly sets the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` annotation on the default router Service when the corresponding flag is enabled in the IngressController spec.
+
+**installer**:
+
+- Implement a job (dedicated or as part of existing e2e suites) to exercise the e2e flow of enabling Security Group management for NLB during cluster installation by setting the platform.aws.`ingressController.managedSecurityGroup` option in `install-config.yaml`.
+
+> Do we really need a new cluster or just a job rendering the install-config.yaml, then later reuse an existing cluster to create a new ingress controller with rendered manifest?
+
+**API**:
+
+- Unit tests for the new boolean field `managedSecurityGroup` in the IngressController CRD.
+
+## Graduation Criteria
+
+> TODO/TBD
+
+### Dev Preview -> Tech Preview
+
+N/A. This feature will be introduced as Tech Preview.
+
+### Tech Preview -> GA
+
+The E2E tests should be consistently passing and a PR will be created to enable the feature gate by default.
+
+### Removing a deprecated feature
+
+> TODO/TBD: depends on the options.
+
+- Announce deprecation and support policy of the existing lbType configuration (if accepted)
+- Deprecate the field
+
+## Upgrade / Downgrade Strategy
+
+> TODO/TBD: depends on the options.
+
+No upgrade or downgrade* concerns because all changes are compatible or in the installer.
+
+There is no goal to migrate existing Routers with NLB to use Security Group, instead, Day-2 operations must be manually added if the user wants to consume this feature.
```

Comment:
i think it might be worth it to mention again that the SG cannot be applied after the NLB is created, so any migration must involve a recreation of the service.

Reply:
Good catch. it has been fixed in the latest version.